controllers/usersignup/usersignup_controller_test.go

```diff
@@ -4717,32 +4760,83 @@ func TestUserSignupStatusNotReady(t *testing.T) {
+		r, req, _ := prepareReconcile(t, userSignup.Name, append(initObjects, userSignup, mur, space, spacebinding)...)
+
 		// when
 		res, err := r.Reconcile(context.TODO(), req)
+
+		// then
 		require.NoError(t, err)
 		require.Equal(t, reconcile.Result{}, res)
-		// then
 		err = r.Client.Get(context.TODO(), types.NamespacedName{Name: userSignup.Name, Namespace: req.Namespace}, userSignup)
 		require.NoError(t, err)
 		test.AssertConditionsMatch(t, userSignup.Status.Conditions, signupComplete...)
+
+		t.Run("the same for mur", func(t *testing.T) {
+			// given
+			setSpaceToReady(t, r.Client, space.Name)
+			mur.Status.Conditions = []toolchainv1alpha1.Condition{{
+				Type:   toolchainv1alpha1.ConditionReady,
+				Status: corev1.ConditionFalse,
+				Reason: toolchainv1alpha1.MasterUserRecordProvisioningReason,
+			}}
+			require.NoError(t, r.Client.Update(context.TODO(), mur))
+
+			// when
+			res, err := r.Reconcile(context.TODO(), req)
+
+			// then
+			require.NoError(t, err)
+			require.Equal(t, reconcile.Result{}, res)
+			err = r.Client.Get(context.TODO(), types.NamespacedName{Name: userSignup.Name, Namespace: req.Namespace}, userSignup)
+			require.NoError(t, err)
+			test.AssertConditionsMatch(t, userSignup.Status.Conditions, signupComplete...)
+
+		})
 	})
 
-	// If a space is updating, usersignups that haven't already completed should be marked as incomplete
+	// If either space is updating or mur is not ready, usersignups that haven't already completed should be marked as incomplete
 	t.Run("mark in-progress signups as incomplete when space is updating", func(t *testing.T) {
 		// given
 		userSignup, mur, space, spacebinding := setup()
-		space.Status.Conditions = signupIncomplete
-		userSignup.Status.Conditions = []toolchainv1alpha1.Condition{}
-		r, req, _ := prepareReconcile(t, userSignup.Name, spc1, userSignup, mur, space, spacebinding, commonconfig.NewToolchainConfigObjWithReset(t, testconfig.AutomaticApproval().Enabled(true)), baseNSTemplateTier, deactivate30Tier)
+		userSignup.Status.Conditions = signupIncomplete("mur")
+		space.Status.Conditions = []toolchainv1alpha1.Condition{{
+			Type:   toolchainv1alpha1.ConditionReady,
+			Status: corev1.ConditionFalse,
+			Reason: toolchainv1alpha1.SpaceUpdatingReason,
+		}}
+		r, req, _ := prepareReconcile(t, userSignup.Name, append(initObjects, userSignup, mur, space, spacebinding)...)
+
 		// when
 		res, err := r.Reconcile(context.TODO(), req)
+
+		// then
 		require.NoError(t, err)
 		require.Equal(t, reconcile.Result{}, res)
-		// then
 		err = r.Client.Get(context.TODO(), types.NamespacedName{Name: userSignup.Name, Namespace: req.Namespace}, userSignup)
 		require.NoError(t, err)
-		test.AssertConditionsMatch(t, userSignup.Status.Conditions, signupIncomplete...)
+		test.AssertConditionsMatch(t, userSignup.Status.Conditions, signupIncomplete("space")...)
```

Comment:
Is this test and the subtest different compared to "until Space is provisioned" and  "when space is provisioned, but not mur" , they seem to be checking the same conditions, but I might have missed something.

Reply:
good point, they are technically the same. The only diff is that the Space & MUR have specific condition set in the second case, but it doesn't really matter if they don't have any condition at all, or some specific set - in both cases they are not ready. 
removed the second test in 97eeacf9